src/test/ui/if/if-no-match-bindings.stderr

```diff
@@ -20,7 +26,10 @@ error[E0308]: mismatched types
   --> $DIR/if-no-match-bindings.rs:20:8
    |
 LL |     if &true {}
-   |        ^^^^^ expected bool, found &bool
+   |        ^^^^^
+   |        |
+   |        expected bool, found &bool
+   |        help: consider dereferencing the borrow: `*&true`
```

Comment:
This tells me that `suggest_ref_or_into` needs to check for literal borrows before suggesting, but that can be done in a subsequent PR.